Code/Framework/AzToolsFramework/AzToolsFramework/UI/DocumentPropertyEditor/DocumentPropertyEditor.cpp

```diff
@@ -212,7 +311,23 @@ namespace AzToolsFramework
                     subPath.Pop();
                 }
                 const auto valueAtSubPath = GetDPE()->GetAdapter()->GetContents()[subPath];
-                m_widgetToPropertyHandler[childWidget]->SetValueFromDom(valueAtSubPath);
+
+                // check if it's a PropertyHandler; if it is, just set it from the DOM directly
+                auto foundEntry = m_widgetToPropertyHandler.find(childWidget);
+                if (foundEntry != m_widgetToPropertyHandler.end())
+                {
+                    foundEntry->second->SetValueFromDom(valueAtSubPath);
+                }
+                else
+                {
+                    QLabel* changedLabel = qobject_cast<QLabel*>(childWidget);
+                    AZ_Assert(changedLabel, "not a label, unknown widget discovered!");
+                    if (changedLabel)
+                    {
+                        auto labelString = valueAtSubPath[AZ::DocumentPropertyEditor::Nodes::Label::Value.GetName()].GetString();
```

Comment:
Same as above w/ tricksy no label text labels. Not for this PR, but if we end up supporting other attributes that affect label (like color or something for overrides) it might be worth just having a general `ReadLabelAttributesFromDom` method to DRY this.

Reply:
done, thanks!